### ISSUE TITLE
2.9.0: Cross-Compile w/ MPI & SST

### ISSUE THREAD

**Issue by ax3l** [ornladios/ADIOS2#3647]
**Describe the bug**
This is a Python regression in 2.9.0, first seen by @liangwang0734 in https://github.com/conda-forge/adios2-feedstock/pull/71

cc @vicentebolea - thank you for offering to help with this

**To Reproduce**
Build and test ADIOS2 as in https://github.com/conda-forge/adios2-feedstock/pull/71 for cross-compilation from x86 to arm64 macOS builds.

The issue we see is:
```
CMake Error: TRY_RUN() invoked in cross-compiling mode, please set the following cache variables appropriately:
  ADIOS2_HAVE_MPI_CLIENT_SERVER_EXITCODE (advanced)
  ADIOS2_HAVE_MPI_CLIENT_SERVER_EXITCODE__TRYRUN_OUTPUT (advanced)
Performing Test ADIOS2_HAVE_MPI_CLIENT_SERVER - Failed
```

**Desktop (please complete the following information):**
 - OS/Platform: OSX x86_64 **cross-compiling for arm64**
 - Build: **LLVM** Clang 14, CMake (latest), shared build

**Additional context**
This bug prevents us from releasing a new ADIOS2 release on conda-forge.

**Comment by ax3l** [comment]
I suspect the way to work around this is to set an ADIOS2 CMake cache variable explicitly...

Generally, is there a way to express this without `try_run` in ADIOS2's CMake logic? Note that `try_run` is generally problematic for cross-compilation, since it would require that you have a way to emulate running the built binaries at compile-time.

**Comment by ax3l** [comment]
Digging deeper, this seems to be an SST issue cc @eisenhauer 

Is is possible that SST only supports MPICH now and no OpenMPI?
https://github.com/ornladios/ADIOS2/blob/v2.9.0/cmake/DetectOptions.cmake#L398-L431

**Comment by eisenhauer** [comment]
While it is SST related, it's the MPI data plane and that's @vicentebolea 's baby, so I'll punt to him.

**Comment by vicentebolea** [comment]
@ax3l you can enable/disable the check and the feature directly with `-DADIOS2_HAVE_MPI_CLIENT_SERVER=ON/OFF`.

On the long term, we need to fix this. Thanks for reporting @ax3l

**Comment by vicentebolea** [comment]
>  Is there a way to find out ADIOS2_HAVE_MPI_CLIENT_SERVER without a runtime test?

No, the reason of using a runtime test is that we cannot know for sure that the mpi library supports clients/server without actually trying a sample code. This is due to the flaky support and its dependence for having the mpi library to use the "right" build/runtime flags.

We added this to stop having users that did not intended to use the MPI dataplane have their sst failing due to this.

Closing this as the there is not solution for this other than using the `-DADIOS2_HAVE_MPI_CLIENT_SERVER=ON/OFF` in a cross compilation.
